File: PN15.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void Display()      
        {
            cout<<"Value is : "<<this->iNo<<endl;
        }

        void DisplayNonFactors()
		{
			int iCnt = 0;
			
			for(iCnt = 1;iCnt < (iNo/2); iCnt++)
			{
				if((iNo % iCnt) != 0)
				{
					cout<<iCnt<<endl;
				}
			}
		
		}

};

int main()
{
    Number nobj;

    nobj.Accept();
    nobj.Display();

    nobj.DisplayNonFactors();

    return 0;
}

File: PN15.java
//Reverse

import java.lang.*;
import java.util.*;

class Number
{
	private int iNo;
	
	public void Accept()
	{
		Scanner sobj = new Scanner(System.in);
		System.out.println("Enter number : ");
		this.iNo = sobj.nextInt();
	}
	
	public void Display()
	{
		System.out.println("Value is: "+this.iNo);
	}
	
	public void Reverse()
	{
		int iDigit = 0;
		int iRev = 0;
		
		while(iNo > 0)
		{
			iDigit = iNo %10;
			iRev = iRev * 10 + iDigit;
			iNo = iNo/10;
		}
		System.out.println(iRev);
	}
	
}

class PN15
{
	public static void main(String x[])
	{
		Number nobj = new Number();
		
		nobj.Accept();
		nobj.Display();
		
		nobj.Reverse();
	}
}